### PR TITLE
Adds draft and past sub route

### LINKED ISSUE #149: Add past to events route.(UI only)
**I'm submitting a ...**  (check one with "x")
- [ ] bug report
- [x] feature request
- [ ] support request => Please do not submit support requests here, instead ask your query in out Gitter channel at https://gitter.im/fossasia/open-event-frontend

Add past as sub route of events.

alt text

located at events/past

### PR DESCRIPTION
**Pull request by anu-007**
#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

#### Short description of what this resolves:
Adds past sub route

#### Changes proposed in this pull request:

alt text

demo- ( https://open-event-frontend-pr-for-149.herokuapp.com/events/past )

Fixes #149 #143

### CODE REVIEW

File: app/controllers/events/past.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+      propertyName : 'url',
+      title        : 'Public URL',
+      class        : 'hide'
+    },
+    {
+      template : 'components/ui-table/cell/cell-buttons',
+      title    : 'Actions'
+    }
+  ],
+  data: [{
```
Comment (geekyd) [inline app/controllers/events/past.js]: Move this to route file

Comment (anu-007) [inline app/controllers/events/past.js]: it will be handled by [#155](https://github.com/fossasia/open-event-frontend/pull/155)

Comment (geekyd) [inline app/controllers/events/past.js]: @anu-007 We have #155 for moving dummy data from controllers to routes that was introduced in early commits not for new additions. Please add this to the respective route file.

File: app/templates/components/ui-table/cell/cell-sessions.hbs [hunk 2]
```diff
@@ -1,8 +1,8 @@
 <div class="ui list">
-  <div class="item">Drafts: 0</div>
-  <div class="item">Submitted: 0</div>
-  <div class="item">Accepted: 0</div>
-  <div class="item">Confirmed: 0</div>
-  <div class="item">Pending: 0</div>
-  <div class="item">Rejected: 0</div>
+  <div class="item">{{t 'Drafts'}}: {{record.sessions.[0]}}</div>
```
Comment (geekyd) [inline app/templates/components/ui-table/cell/cell-sessions.hbs]: Please revert this. We will take care of improving the cell templates in #161

File: app/templates/components/ui-table/cell/cell-tickets.hbs [hunk 3]
```diff
@@ -1 +1 @@
-<div class="ui bulleted list"><div class="item">Premium (12/100)</div><div class="item">VIP (10/15)</div><div class="item">Normal (100/200)</div><div class="item">Free (100/500)</div></div>
```
Comment (geekyd) [inline app/templates/components/ui-table/cell/cell-tickets.hbs]: Please revert this. We will take care of improving the cell templates in #161

**Comment by anu-007** [comment]
@geekyd please review

File: app/templates/events/past.hbs [hunk 4]
```diff
@@ -1 +1 @@
-Past data
```
Comment (niranjan94) [inline app/templates/events/past.hbs]: Please don't create a new template. Just reuse the same template as `live`

File: app/templates/components/ui-table/cell/cell-roles.hbs [hunk 5]
```diff
@@ -1 +1 @@
-<span>sample@gmail.com (Organizer)</span>
+<span>{{record.roles}}(Organizer)</span>
```
Comment (geekyd) [inline app/templates/components/ui-table/cell/cell-roles.hbs]: Do not manipulate this file. It will be taken care of #161

**Comment by niranjan94** [comment]
Okay .. this comment is in reference to both #156 and #148 

Draft, live and past are the exact same with the exception of the UI. So please make all three pages into a single route using [dynamic segments](https://guides.emberjs.com/v2.13.0/routing/defining-your-routes/#toc_dynamic-segments)

**Comment by anu-007** [comment]
@niranjan94 what should I use as the dynamic part of the URL.

**Comment by niranjan94** [comment]
@anu-007 the dynamic part would be `live`, `draft`, `past` 

`/events/:event_state`

**Comment by anu-007** [comment]
@niranjan94 templates are not rendered properly using dynamic segments can you help me in this.thanks

**Comment by mariobehling** [comment]
@anu-007 Just to be sure, we all understand the role of mentors. Mentors are not teachers. At FOSSASIA we don't have enough resources to provide trainings during GSoC. I see that a number of students continuously ask for help. Asking for help is fine, but I would expect you do more research first and work with other students first before asking for mentors support. Please be savy with the time of mentors. Mentors are volunteers and don't get paid for work during GSoC as opposed to students.

Please speed up your reactions to comments in PRs and show us that you can actually achieve quality commits without many comments of mentors. We are now in the coding phase of GSoC and expectations are higher than in the bonding period. Please provide quality commits now and speed up the work on PRs to get them ready for merges on the same day where-ever possible.

**Comment by niranjan94** [comment]
@anu-007 I think you need to read up on how routes work in ember. especially dynamic one's as we'll be needing to use them in the app. 
- As a first step, close PR #148 .
- Rename `live` route/template to `list`
- Delete draft and past routes and template
- In router.js ensure, `this.route('live');` is changed to `this.route('list', { path: '/:event_state' });`

Now going to all routes `/events/live`, `/events/draft`, `/events/past` is gonna display the table for you 

Now, you can work from this point.

**Comment by anu-007** [comment]
@niranjan94 done changes as you said but still the templates are not rendered properly also this one is taking too much time, should I close it and work on some other issue.thanks

**Comment by niranjan94** [comment]
@anu-007 
> but still the templates are not rendered properly

Can you tell me what exactly is happening ?

**Comment by niranjan94** [comment]
> this one is taking too much time, should I close it and work on some other issue 

Somebody has to do this right and you do need to learn how to ? And since you took it up, I suggest you don't give up right now ...

**Comment by anu-007** [comment]
the template corresponding to events (events.hbs) is rendering partially i.e. only till the link corresponding to 'list ' and hence the nested templates list,draft,past are not rendered.

**Comment by anu-007** [comment]
I have pushed the updated code you can have a look and tell me where i was doing it wrong.thanks

File: app/templates/events.hbs [hunk 6]
```diff
@@ -2,7 +2,7 @@
   <div class="row">
     <div class="sixteen wide column">
       {{#tabbed-navigation}}
-        {{#link-to 'events.live' class='item'}}
+        {{#link-to 'events.list' class='item'}}
           {{t 'Live'}}
         {{/link-to}}
         {{#link-to 'events.draft' class='item'}}
```
Comment (niranjan94) [inline app/templates/events.hbs]: These links haven't been changed yet

File: app/templates/components/nav-bar.hbs [hunk 7]
```diff
@@ -31,7 +31,7 @@
           <div class="menu">
             <a href="{{href-to 'my-tickets'}}" class="item">{{t 'My Tickets'}}</a>
             <div class="item">{{t 'My Sessions'}}</div>
-            <a href="{{href-to 'events.live'}}" class="item">{{t 'Manage Events'}}</a>
+            <a href="{{href-to 'events.list'}}" class="item">{{t 'Manage Events'}}</a>
```
Comment (niranjan94) [inline app/templates/components/nav-bar.hbs]: Should be `{href-to 'events.list' 'live'}}`

File: app/templates/components/side-bar.hbs [hunk 8]
```diff
@@ -13,7 +13,7 @@
     </div>
     <a href="{{href-to 'my-tickets'}}" class="item">{{t 'My Tickets'}}</a>
     <div class="item">{{t 'My Sessions'}}</div>
-    <a href="{{href-to 'events.live'}}" class="item">{{t 'Manage Events'}}</a>
+    <a href="{{href-to 'events.list'}}" class="item">{{t 'Manage Events'}}</a>
```
Comment (niranjan94) [inline app/templates/components/side-bar.hbs]: Should be `{href-to 'events.list' 'live'}}`

File: app/templates/components/nav-bar.hbs (hunk 7 above)
Comment (niranjan94) [inline app/templates/components/nav-bar.hbs]: Should be `{href-to 'events.list' 'live'}}`

File: app/templates/events.hbs (hunk 6 above)
Comment (niranjan94) [inline app/templates/events.hbs]: Should be 

```hbs
        {{#link-to 'events.list' 'live' class='item'}}
          {{t 'Live'}}
        {{/link-to}}
        {{#link-to 'events.list' 'draft' class='item'}}
          {{t 'Draft'}}
        {{/link-to}}
        {{#link-to 'events.list' 'past' class='item'}}
          {{t 'Past'}}
        {{/link-to}}
```

**Comment by anu-007** [comment]
@niranjan94 @geekyd please review ( https://pr-156-open-event-fronend.herokuapp.com/events/live ).thanks

**Comment by anu-007** [comment]
@geekyd @niranjan94 updated branch please review

File: app/routes/events/index.js [hunk 9]
```diff
@@ -5,6 +5,6 @@ const { Route } = Ember;
 export default Route.extend({
   beforeModel() {
     this._super(...arguments);
-    this.transitionTo('events.live');
+    this.transitionTo('events.list');
```
Comment (niranjan94) [inline app/routes/events/index.js]: Should be `this.transitionTo('events.list', 'live')`

**Comment by anu-007** [comment]
@niranjan94 please review

**Comment by anu-007** [comment]
@niranjan94 updated branch please review thanks

**Pull request merged**